From 47e6b57867dc7a1717bc8594c837287aa5641333 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 15 Nov 2018 12:45:17 +0530
Subject: [PATCH 0487/1852] drm: xlnx: dsi: Fix kernel panic when "halt" is run

commit 1f1aaea8dba55042e3d0746f8dbf5798b1f103d3 from
https://github.com/Xilinx/linux-xlnx.git

When "halt" is run, the kernel panics and dumps a kernel call stack
which showing it originating in xlnx_dsi_unbind(). On closer
examination, it is found that the xlnx_dsi_host_detach() incorrectly
sets the "panel_mode" device node member pointer to NULL instead of
setting the drm_panel member pointer "panel" to NULL.

The patch corrects this incorrect assignment.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Sandip Kothari <sandipk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_dsi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_dsi.c b/drivers/gpu/drm/xlnx/xlnx_dsi.c
index 4edd5d27d6b5..02561fd1b300 100644
--- a/drivers/gpu/drm/xlnx/xlnx_dsi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_dsi.c
@@ -405,7 +405,7 @@ static int xlnx_dsi_host_detach(struct mipi_dsi_host *host,
 {
 	struct xlnx_dsi *dsi = host_to_dsi(host);
 
-	dsi->panel_node = NULL;
+	dsi->panel = NULL;
 
 	if (dsi->connector.dev)
 		drm_helper_hpd_irq_event(dsi->connector.dev);
-- 
2.31.1

